src/modelframe.jl

```diff
@@ -1,3 +1,50 @@
+"""
+Wrapper which combines Formula (Terms) and an AbstractDataFrame
+
+This wrapper encapsulates all the information that's required to transform data
+of the same structure as the wrapped `DataFrame` into a model matrix.  This goes
+above and beyond what's expressed in the `Formula` itself, for instance
+including information on how each categorical variable should be coded.
+
+Creating a `ModelFrame` first parses the `Formula` into `Terms`, checks which
+variables are categorical and determines the appropriate contrasts to use, and
+then creates the necessary contrasts matrices and stores the results.
+
+**Constructors**
+
+```julia
+ModelFrame(f::Formula, df::AbstractDataFrame; contrasts::Dict = Dict())
+ModelFrame(ex::Expr, d::AbstractDataFrame; contrasts::Dict = Dict())
+ModelFrame(terms::Terms, df::AbstractDataFrame; contrasts::Dict = Dict())
+# Internal constructors:
+ModelFrame(df::AbstractDataFrame, terms::Terms, missing::BitArray)
+ModelFrame(df::AbstractDataFrame, terms::Terms, missing::BitArray, contrasts::Dict{Symbol, ContrastsMatrix})
+```
+
+**Arguments**
+
+* `f`: Formula whose left hand side is the _response_ and right hand side are
```

Comment:
Also maybe repeat the argument types here and below, this often clarifies a lot what it's supposed to contain.